pandas/sparse/frame.py

```diff
@@ -187,9 +213,47 @@ def _init_matrix(self, data, index, columns, dtype=None):
         if len(index) != N:
             raise ValueError('Index length mismatch: %d vs. %d' %
                              (len(index), N))
+        return index, columns
 
-        data = dict([(idx, data[:, i]) for i, idx in enumerate(columns)])
-        return self._init_dict(data, index, columns, dtype)
+    def to_coo(self):
+        """
+        Convert the frame to its SciPy sparse COO matrix representation.
```

Comment:
SciPy matrices are indexed (rows and columns) sequentially 0..(N-1), 0..(M-1), like ndarrays. Obviously, none of the DF indices are retained on the coo_matrix, only the raw values in the same shape. I don't know what to write. :blush: 

Perhaps I should change "Convert the frame ..." to "Return a SciPy sparse matrix ..." as nothing happens in-place.